samza-core/src/test/java/org/apache/samza/container/TestRunLoop.java

```diff
@@ -236,7 +236,7 @@ public void testProcessMultipleTasks() throws Exception {
     TaskInstance t0 = createTaskInstance(task0, taskName0, ssp0, offsetManager, consumerMultiplexer);
     TaskInstance t1 = createTaskInstance(task1, taskName1, ssp1, offsetManager, consumerMultiplexer);
 
-    Map<TaskName, TaskInstance> tasks = new HashMap<>();
```

Comment:
Should `RunLoopTask` now be mocked in this test instead of using a concrete `TaskInstance`? It could help to validate your extraction of the interface and simplify the test to not depend on `TaskInstance`.

Reply:
Removed `TaskInstance` and instead had `TestTask` be an implementer of `RunLoopTask` interface.